src/cloudWatchLogs/document/logStreamDocumentProvider.ts

```diff
@@ -67,12 +68,24 @@ export class LogStreamDocumentProvider implements vscode.TextDocumentContentProv
             logGroupInfo.streamName = streamID
             const initialStreamData = getInitialLogData(logGroupInfo, parameters, getLogEventsFromUriComponents)
             const streamUri = createURIFromArgs(logGroupInfo, parameters)
+
             await this.registry.registerLog(streamUri, initialStreamData)
-            // Set the document language
             const doc = await vscode.workspace.openTextDocument(streamUri)
             vscode.languages.setTextDocumentLanguage(doc, 'log')
+
+            telemetry.recordCloudwatchlogsOpen({
```

Comment:
Is this redundant with the `recordCloudwatchlogsOpen` metric that will be emitted when the logstream is loaded? Can the logstream viewer set `source=goto` based on the URI (maybe we could add a `?source=goto` querystring or something like that?

Reply:
The `recordCloudwatchlogsOpen` code is only run in the `viewLogStream` function which isn't called in the jumpTo code. There is one issue where `recordCloudwatchlogsOpen` is recorded twice (once on the hover of the text, and once on the loading of the new file). It seems like both actions run the `provideDefinition` code, and I haven't found a way to distinguish them.